tests/components/tuya/conftest.py

```diff
@@ -142,11 +143,11 @@ async def mock_device(hass: HomeAssistant, mock_device_code: str) -> CustomerDev
     device.product_name = details["product_name"]
     device.online = details["online"]
     device.function = {
-        key: MagicMock(type=value["type"], values=value["values"])
+        key: MagicMock(type=value["type"], values=json_dumps(value["value"]))
```

Comment:
I have adjusted this to match what diagnostic dumps provide.

See dump from https://github.com/home-assistant/core/issues/144745 (for `cwwsq`) and https://github.com/home-assistant/core/pull/146599 (for `cwysj`)

cc @FredrikM97